onnxscript/ir/_external_data.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Pass to save tensor data as external tensors."""
+
+import os
+
+from onnxscript.ir import _core
+
+
+class ExternalDataInfo:
+    def __init__(self, name, offset, length):
+        self.name = name
+        self.offset = offset
+        self.length = length
+
+
+def save_external_data(
+    initializers,
+    file_path,
+    allocation_granularity: int = 65536,  # 64KB
+):
```

Comment:
I would add a docstring and document what happens when the data file does exist (overwriten)